hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/jdbc/JdbcSqlConnector.java

```diff
@@ -156,6 +161,23 @@ private Map<String, DbField> readDbFields(
         }
     }
 
+    @Override
+    @Nonnull
+    public List<Permission> permissionsForResolve(SqlExternalResource resource,
+                                                  NodeEngine nodeEngine) {
+        JdbcDataConnection jdbcDataConnection = nodeEngine.getDataConnectionService()
+                                                          .getAndRetainDataConnection(resource.dataConnection(),
```

Comment:
do we really need to connect to the database just to get it's URL?

Reply:
I intentionally did this so it's in sync with 
```
https://github.com/hazelcast/hazelcast/blob/3dbc936c5fc22e6ab6f761df6439295156af0511/hazelcast/src/main/java/com/hazelcast/jet/impl/connector/ReadJdbcP.java#L143-L146
```
We could read the data connection config and check the `jdbcUrl` parameter as well. But it won't save much, if the check is successful we create the connection and read the fields just after this check. So it only saves anything in case of failed check.

@kwart wdyt?

Reply:
but you did not invoke `DriverManager.getDrivers();` which according to those comments might lead to deadlocks

Reply:
> But it won't save much, if the check is successful we create the connection and read the fields just after this check. So it only saves anything in case of failed check.

It won't be much difference if we use connection pooling, which is used only for shared DCs. For non-shared we need 2 connections. However, this should not be that important, `CREATE MAPPING` should not be invoked too often. Still, IMO it would be good to add a comment about this.

Reply:
I would try to avoid establishing DB connections before they are used. But I have no strong objection here.